dialects/terraform/src/scanner.c

```diff
@@ -163,16 +163,17 @@ static void deserialize(Scanner *scanner, const char *buffer, unsigned length) {
 
     VEC_CLEAR(scanner->context_stack);
     unsigned size = 0;
-    uint8_t context_stack_size = buffer[size++];
+    uint32_t context_stack_size;
+    memcpy(&context_stack_size, &buffer[size], sizeof(uint32_t));
```

Comment:
i think we are not incrementing size afterwards and it crashes because of that, could that be?

Reply:
ah oops - yeah

Before it was size++ so it was incremented (just truncated)